### DISCUSSION TITLE
Usage in pyroot with RDataFrame

### DISCUSSION THREAD

**Discussion by sparedes** [cms-nanoAOD/correctionlib#101] [Q&A] [answered]
I have an event selection script that relies heavily on `RDataFrame` and it uses pyroot. 
I'm trying to use `correctionlib` to get scale factors using `cms-nanoAOD/jsonpog-integration`, following something like [this example](https://gist.github.com/pieterdavid/a560e65658386d70a1720cb5afe4d3e9#file-df-py) linked from the README. Here is a test example on an interactive python session:
```
>>> import ROOT
>>> import correctionlib
>>> correctionlib.register_pyroot_binding()
>>> ROOT.gInterpreter.Declare('auto e_SF_map = correction::CorrectionSet::from_file("jsonpog-integration/POG/EGM/2017_UL/electron.json");')
IncrementalExecutor::executeFunction: symbol '_ZN10correction13CorrectionSet9from_fileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE' unresolved while linking function '_GLOBAL__sub_I_cling_module_65'!
You are probably missing the definition of correction::CorrectionSet::from_file(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
Maybe you need to load the corresponding shared library?
True
```

I'm guessing there is a step missing somewhere, which means the `gInterpreter` can't find `correction` when doing `correctionlib.register_pyroot_binding()`, or something like this? Are there some other setup instructions I might be missing? 

Using `pip install` and doing something similar to [this example](https://gitlab.cern.ch/cms-nanoAOD/jsonpog-integration/-/blob/master/examples/electronExample.py) works just fine, but I think the `gInterpreter` route would work better for my setup.

**Comment by nsmith-** [comment, marked as answer]
This looks suspiciously like the C++11 ABI incompatibility issue discussed in https://github.com/cms-nanoAOD/correctionlib/discussions/86#discussioncomment-799849 How exactly did you install correctionlib in this environment? Also what is the environment? CMSSW?

**Reply by sparedes** [reply]
To install I just used `pip`
```shell
python3 -m pip install correctionlib --user
```
 And environment-wise I only use an LCG release, in this test I used 
```shell
source /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-slc6-gcc8-opt/setup.sh
```

**Reply by nsmith-** [reply]
Can you try the alternative installation instruction from the other thread? Namely:
`python3 -m pip install --user --no-binary=correctionlib correctionlib`
(after sourcing LCG env)

**Reply by sparedes** [reply]
This worked - thanks!
